library/ceph_pool.py

```diff
@@ -546,66 +640,93 @@ def run_module():
+        rc, cmd, out, err = exec_commands(module,
+                                          check_pool_exist(cluster,
+                                                           name,
+                                                           user,
+                                                           user_key,
+                                                           container_image=container_image))  # noqa: E501
         if rc == 0:
-            for k, v in def_opt.items():
-                if module.params[k] == None:
-                    user_pool_config[k] = {'value': default_running_ceph_config[v['conf_name']], 'cli_set_opt': v['cli_set_opt']}
-                else:
-                    user_pool_config[k] = {'value': module.params.get(k), 'cli_set_opt': v['cli_set_opt']}
-            rc, cmd, out, err = exec_commands(module, check_pool_exist(cluster, name, user, user_key, container_image=container_image))
-            if rc == 0:
-                running_pool_details = get_pool_details(module, cluster, name, user, user_key, container_image=container_image)
-                user_pool_config['pg_placement_num'] = { 'value': str(running_pool_details[2]['pg_placement_num']), 'cli_set_opt': 'pgp_num' }
-                delta = compare_pool_config(user_pool_config, running_pool_details[2])
-                if len(delta) > 0 and running_pool_details[2]['erasure_code_profile'] == "" and 'size' not in delta.keys():
-                    rc, cmd, out, err = update_pool(module, cluster, name, user, user_key, delta, container_image=container_image)
-                    if rc == 0:
-                        changed = True
-                else:
-                    out = "Pool {} already exists and there is nothing to update.".format(name)
+            running_pool_details = get_pool_details(module,
+                                                    cluster,
+                                                    name,
+                                                    user,
+                                                    user_key,
+                                                    container_image=container_image)  # noqa: E501
+            user_pool_config['pg_placement_num'] = {'value': str(running_pool_details[2]['pg_placement_num']), 'cli_set_opt': 'pgp_num'}  # noqa: E501
+            delta = compare_pool_config(user_pool_config,
+                                        running_pool_details[2])
+            if len(delta) > 0:
+                keys = list(delta.keys())
+                details = running_pool_details[2]
+                if details['erasure_code_profile'] and 'size' in keys:
+                    del delta['size']
+                if details['pg_autoscale_mode'] == 'on':
+                    delta.pop('pg_num', None)
+                    delta.pop('pgp_num', None)
+
+                rc, cmd, out, err = update_pool(module,
```

Comment:
TODO: check whether there is an impact when running `update_pool()` in case of `delta` is empty after removing either `size`, `pg_num` or `pgp_num`